cmake/modules/CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+# Generate a list of CMake library targets so that other CMake projects can
+# link against them. LLVM calls its version of this file LLVMExports.cmake, but
+# the usual CMake convention seems to be ${Project}Targets.cmake.
+set(MLIR_INSTALL_PACKAGE_DIR lib${LLVM_LIBDIR_SUFFIX}/cmake/llvm)
+set(mlir_cmake_builddir "${CMAKE_BINARY_DIR}/${MLIR_INSTALL_PACKAGE_DIR}")
+
+# Keep this in sync with llvm/cmake/CMakeLists.txt!
+set(LLVM_INSTALL_PACKAGE_DIR lib${LLVM_LIBDIR_SUFFIX}/cmake/llvm)
+set(llvm_cmake_builddir "${LLVM_BINARY_DIR}/${LLVM_INSTALL_PACKAGE_DIR}")
+
+get_property(MLIR_EXPORTS GLOBAL PROPERTY MLIR_EXPORTS)
+export(TARGETS ${MLIR_EXPORTS} FILE ${mlir_cmake_builddir}/MlirTargets.cmake)
+
+# Generate MlirConfig.cmake for the build tree.
+set(MLIR_CONFIG_CMAKE_DIR "${mlir_cmake_builddir}")
+set(MLIR_CONFIG_LLVM_CMAKE_DIR "${llvm_cmake_builddir}")
+set(MLIR_CONFIG_EXPORTS_FILE "${mlir_cmake_builddir}/MlirTargets.cmake")
+set(MLIR_CONFIG_INCLUDE_DIRS
+  "${MLIR_SOURCE_DIR}/include"
+  "${MLIR_BINARY_DIR}/include"
+  )
+set(MLIR_CONFIG_CMAKE_DIR)
+set(MLIR_CONFIG_LLVM_CMAKE_DIR)
+set(MLIR_CONFIG_EXPORTS_FILE)
+
+# Generate MlirConfig.cmake for the install tree.
+set(MLIR_CONFIG_CODE "
+# Compute the installation prefix from this LLVMConfig.cmake file location.
+get_filename_component(MLIR_INSTALL_PREFIX \"\${CMAKE_CURRENT_LIST_FILE}\" PATH)")
+# Construct the proper number of get_filename_component(... PATH)
+# calls to compute the installation prefix.
+string(REGEX REPLACE "/" ";" _count "${MLIR_INSTALL_PACKAGE_DIR}")
+foreach(p ${_count})
+  set(MLIR_CONFIG_CODE "${MLIR_CONFIG_CODE}
+get_filename_component(MLIR_INSTALL_PREFIX \"\${MLIR_INSTALL_PREFIX}\" PATH)")
+endforeach(p)
+set(MLIR_CONFIG_CMAKE_DIR "\${MLIR_INSTALL_PREFIX}/${MLIR_INSTALL_PACKAGE_DIR}")
+set(MLIR_CONFIG_LLVM_CMAKE_DIR "\${MLIR_INSTALL_PREFIX}/${LLVM_INSTALL_PACKAGE_DIR}")
+set(MLIR_CONFIG_EXPORTS_FILE "\${MLIR_CMAKE_DIR}/MlirTargets.cmake")
+set(MLIR_CONFIG_INCLUDE_DIRS
+  "\${MLIR_INSTALL_PREFIX}/include"
+  )
+#configure_file(
+#  ${CMAKE_CURRENT_SOURCE_DIR}/MlirConfig.cmake.in
+#  ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/MlirConfig.cmake
+#  @ONLY)
```

Comment:
It appears the convention for projects in to have this config file. It's not catastrophic that it isn't present at the moment, but it ought to be added into the mix.